packages/abi-gen/test-cli/test_typescript/test/abi_gen_dummy_test.ts

```diff
@@ -81,22 +80,34 @@ describe('AbiGenDummy Contract', () => {
     });
     describe('simpleRevert', () => {
         it('should call simpleRevert', async () => {
-            expect(abiGenDummy.simpleRevert.callAsync()).to.be.rejectedWith('SIMPLE_REVERT');
+            abiGenDummy.simpleRevert.callAsync().catch(err => {
+                const expected = new StringRevertError('SIMPLE_REVERT');
+                expect(err).to.deep.equal(expected);
+            });
```

Comment:
I'd really like to see us keep the the `expect(...).to.be.rejectedWith(...)` syntax, and I was _sure_ it was possible, but it sure took me a while to figure it out!  How do you feel about switching it to this format?

```
            expect(abiGenDummy.simpleRevert.callAsync())
                .to.be.rejectedWith(StringRevertError)
                .deep.equal(new StringRevertError('SIMPLE_REVERT'));
```

Reply:
Nice! I tried a couple things but gave up early, thanks for taking the time. I think this is also better because it fails sequentially. The way I had it might show a green check mark for a test that actually failed, and then throw an error anyway.